solc/CommandLineInterface.cpp

```diff
@@ -1191,8 +1192,11 @@ void CommandLineInterface::outputCompilationResults()
 	{
 		if (!m_options.output.dir.empty())
 			sout() << "Compiler run successful. Artifact(s) can be found in directory " << m_options.output.dir << "." << endl;
-		else
-			serr() << "Compiler run successful, no output requested." << endl;
+		else if (contracts.empty())
+			sout() << "Compiler run successful, no contracts to compile." << endl;
+		else if (contracts.size() > 0)
```

Comment:
This is already implied by the fact that `contracts.empty()` is false.
```suggestion
		else
```

Reply:
If condition removed.